v4/newrelic/transaction.go

```diff
@@ -126,6 +132,15 @@ func (txn *Transaction) SetWebRequest(r WebRequest) {}
 // package middlewares.  Therefore, you probably want to use this only if you
 // are writing your own instrumentation middleware.
 func (txn *Transaction) SetWebResponse(w http.ResponseWriter) http.ResponseWriter {
+	if w == nil {
+		// Accepting a nil parameter makes it easy for consumers to add
+		// a response code to the transaction without a response
+		// writer:
+		//
+		//    txn.SetWebResponse(nil).WriteHeader(500)
+		//
+		w = dummyResponseWriter{}
+	}
```

Comment:
I believe it's the `nrecho` integration that uses `txn.SetWebResponse(nil).WriteHeader(code)`. Those tests fail without this addition.